cmake/FindSphinx.cmake

```diff
@@ -1,25 +1,13 @@
 include(FindPackageHandleStandardArgs)
 
-find_program(
-  SPHINX_EXECUTABLE
-  NAMES sphinx-build
-        sphinx-build-${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}
-  HINTS $ENV{SPHINX_DIR} PATHS /opt/local/ /usr/local/
-                               $ENV{HOME}/Library/Python/2.7/ PATH_SUFFIXES bin
-  DOC "Sphinx documentation generator.")
+set(SPHINX_EXECUTABLE ${PYTHON_EXECUTABLE} -m sphinx)
+set(SPHINX_API_DOC_EXE ${PYTHON_EXECUTABLE} -m sphinx.apidoc)
```

Comment:
This change and the one in the Cython file generate the following CMake log:
```
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.9", minimum required is "3.5") 
-- Found Cython: /usr/bin/python3;-m;cython (found suitable version "0.26.1", minimum required is "0.26") 
-- Found PythonHeaders: /usr/include/python3.6m  
-- Found NumPy: /tikhome/jgrad/.local/lib/python3.6/site-packages/numpy/core/include (found version "1.17.3") 
-- Found Sphinx: /usr/bin/python3;-m;sphinx (found suitable version "2.0.1", minimum required is "1.6.6") 
```
Is there a way to get the absolute path of Sphinx and Cython instead of a list cast to a string (`/usr/bin/python3;-m;sphinx`)?

Reply:
if not, could we instead quote the value of `SPHINX_EXECUTABLE`, `SPHINX_API_DOC_EXE`, `CYTHON_EXECUTABLE` to prevent the cast to string that adds semicolons?

Reply:
These semicolons are only added when printing the "Found XXX" message. They are correctly unpacked when calling the program. If we convert the list to a string inside FindSphinx/FindCython, CMake later tries to call that whole string as the executable and obviously fails.

> Is there a way to get the absolute path of Sphinx and Cython

No. In fact, there might not even be one.

Reply:
let's first get in functional and then pretty, CI is failing...

Reply:
@jngrad what's the issue with the semicolons?